projects/js-packages/publicize-components/src/hooks/use-attached-media/index.js

```diff
@@ -0,0 +1,33 @@
+import { useSelect, useDispatch } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+
+/**
+ * @typedef {Object} AttachedMediaHook
+ * @property {Array} attachedMedia - List of media IDs.
+ * @property {Function} updateAttachedMedia - Callback used to update the attached media..
+ */
+
+/**
+ * Hook to handle storing the attached media.
+ *
+ * @returns {AttachedMediaHook} - An object with the attached media hook properties set.
+ */
+export default function useAttachedMedia() {
+	const { editPost } = useDispatch( editorStore );
+
+	const attachedMedia = useSelect( select => select( 'jetpack/publicize' ).getAttachedMedia() );
+	const currentOptions = useSelect( select =>
+		select( 'jetpack/publicize' ).getJetpackSocialOptions()
+	);
+
+	return {
+		attachedMedia,
+		updateAttachedMedia: function ( ids ) {
```

Comment:
I've been researching this a bit more, and came across [my favourite SO answer on it](https://stackoverflow.com/a/64134704/5923608). I think we probably do want to wrap this function in `useCallback` so `updateAttachedMedia` is stable between renders, otherwise it will invalidate the `useCallbacks` where it's being used.

Reply:
Makes sense! Fixed in [`0c86a13`](https://github.com/Automattic/jetpack/pull/26930/commits/0c86a13512703d5c596af9f5b83619c7dba77e52).